Discovery/Discovery/Presentation/DiscoveryViewModel.swift

```diff
@@ -20,6 +22,11 @@ public class DiscoveryViewModel: ObservableObject {
     @Published var courses: [CourseItem] = []
     @Published var showError: Bool = false
     
+    var userloggedIn: Bool {
+        guard let container = Container.shared.resolve(CoreStorage.self) else { return false }
```

Comment:
If you need CoreStorage, please inject it in the initializer, this will be best practice to manage dependency.